store/src/lmdb.rs

```diff
@@ -396,7 +396,7 @@ where
 {
 	fn deser_if_prefix_match(&self, key: &[u8], value: &[u8]) -> Option<(Vec<u8>, T)> {
 		let plen = self.prefix.len();
-		if plen == 0 || key[0..plen] == self.prefix[..] {
+		if plen == 0 || (key.len() >= plen && key[0..plen] == self.prefix[..]) {
```

Comment:
What was this change needed for?

Reply:
Oh I get it now after rereading it. We just check the length for safety so it doesn't panic.

Reply:
Yes. Actually I saw a panic here on a test, don't know exact reason, but not bad to give a safe protection here.